Dockerfile

```
@@ -0,0 +1,94 @@
+# syntax=docker/dockerfile:1.5
+# Multi-stage build for crawler_mcp with integrated webhook server
+FROM python:3.11-slim as builder
+
+WORKDIR /app
+
+# Copy project files
+COPY pyproject.toml uv.lock ./
+
+# Install uv and sync dependencies with cache mounts
+RUN --mount=type=cache,target=/root/.cache/pip \
```

Hadolint (2.12.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)